src/test/java/io/vertx/test/core/Http1xTest.java

```diff
@@ -3542,6 +3531,49 @@ public void testPerPeerPoolingWithProxy() throws Exception {
     testPerPeerPooling(i -> client.get(80, "host" + i, "/somepath"));
   }
 
+  @Test
+  public void testNoNPEWhenClientBreaksConnection() throws Exception {
+
+    final File f = setupFile("file.pdf", TestUtils.randomUnicodeString(1000000));
+
+    server.requestHandler(req -> {
+      vertx.executeBlocking(future -> {
```

Comment:
instead you can use a `closeHandler` on the associated `HttpConnection`, that will be more reliable